src/Bridge/Laravel/Providers/HttpClientServiceProvider.php

```diff
@@ -28,10 +29,10 @@ public function register(): void
         $this->app->singleton(ExceptionHandlerInterface::class, ExceptionHandler::class);
 
         // Concrete implementations
-        $this->app->singleton(Client::class);
-        $this->app->singleton(LoggingClient::class, static function (Container $app) {
+        $this->app->singleton(ClientInterface::class, Client::class);
```

Comment:
This wont work, if the application replaces the ClientInterface binding with LoggingClient (which most apps will) it will cause an infinite recursion

Reply:
Ok, we need to be able to replace ClientInterface though, for the purposes of testing though.

What if replaced this
`$this->app->singleton(LoggingClient::class...`
with
`$this->app->singleton('loggingclient'...`
And added the appropriate bindings for `'loggingclient'` ?